scripts/check_changes_in_right_version.sh

```diff
@@ -12,15 +12,15 @@ IFS=$'\n' MODIFIED_FILES=($(git diff --no-commit-id --name-only -r  origin/main.
 for FILE in "${MODIFIED_FILES[@]}"; do
   echo "file is $FILE"
   # if file has the pattern src/modules/ check that is just for CURRENT_VERSION
-  if  [[ "$FILE" == src/modules* ]]; then
-    echo "check version"
+  if  [[ "$FILE" == src/modules/* ]]; then
     # need to check the version
     VERSION=$(echo $FILE|cut -d '/' -f 3)
-    echo "version is $VERSION"
-    if [ "$VERSION" != "$CURRENT_VERSION" ]; then
-      # incorrect version
-      echo "Error, out of version modifications found in file $FILE"
-      exit 1
+    if [[ "$VERSION" =~ ^[0-9]+(\.[0-9]+){2,3}$ ]]; then
```

Comment:
why is the group expected to happen `{2,3}`? should not be only {2} to be a valid semver format?

Reply:
we could also use one of the official regex but they are awful https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string